site/documentation/content/api/command-router/index.md

```diff
@@ -0,0 +1,174 @@
+---
+title: "Command Router API Specification"
+linkTitle: "Command Router API"
+weight: 425
+resources:
+  - src: preconditions.svg
+  - src: register_cmd_consumer.svg
+  - src: unregister_cmd_consumer.svg
+  - src: set_last_known_gateway_success.svg
+---
+
+The *Command Router API* is used by *Protocol Adapters* to configure the Command Router service component, enabling it to route command & control messages to the protocol adapters that the target devices are connected to.
+<!--more-->
+
+The Command Router API is defined by means of AMQP 1.0 message exchanges, i.e. a client needs to connect to Hono using an AMQP 1.0 client in order to invoke operations of the API as described in the following sections.
+
+<a name="preconditions"></a>
+## Preconditions for invoking the Command Router API
+
+1. Client has established an AMQP connection with the Command Router service.
+1. Client has established an AMQP link in role *sender* on the connection using target address `cmd_router/${tenant_id}`. This link is used by the client to send request messages to the Command Router service.
+1. Client has established an AMQP link in role *receiver* on the connection using source address `cmd_router/${tenant_id}/${reply-to}` where *reply-to* may be any arbitrary string chosen by the client. This link is used by the client to receive responses to the requests it has sent to the Command Router service. This link's source address is also referred to as the *reply-to* address for the request messages.
+
+{{< figure src="preconditions.svg" alt="A client establishes an AMQP connection and the links required to invoke operations of the Command Router service" title="Client connecting to Command Router service" >}}
+
+## Register command consumer for device
+
+Clients use this command to *register* a consumer for command & control messages for a device, providing the protocol adapter instance that the given device or gateway is currently connected to.
```

Comment:
This sounds like the consumer and the protocol adapter instance are different things here. How about
> Clients use this command to *register* a protocol adapter instance as the consumer of command & control messages for a device or gateway that is currently connected to that particular adapter instance.